pkg/syncer/fullsync.go

```diff
@@ -240,7 +250,7 @@ func fullSyncDeleteVolumes(ctx context.Context, volumeIDDeleteArray []cnstypes.C
 					log.Warnf("FullSync: fullSyncDeleteVolumes: Failed to delete volume %s with error %+v", volume.VolumeId.Id, err)
 					continue
 				}
-				if metadataSyncer.coCommonInterface.IsFSSEnabled(ctx, common.CSIMigration) {
+				if migrationFeatureStateForFullSync {
```

Comment:
Not related to this change per se, but what if the volume to be deleted was not a migrated volume? There's no need to enter this control path

Reply:
for the successfully deleted volumes, we want to delete CRD for volume path to volume ID mapping.

yes, we need to call `volumeMigrationService.DeleteVolumeInfo` only for migrated volume id, otherwise for non-migrated volume, the call will fail we will never be able to remove the entry from `cnsDeletionMap`.

Reply:
DeleteVolumeInfo for non-migrated volumes will return nil (not error). See here: https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/master/pkg/apis/migration/migration.go#L290
So, the volume id will be deleted from the cnsDeletionMap. I have added appropriate comments to address your comment.

Reply:
We have `apierrors.IsNotFound` logic so we are good.